test/functional/crud_spec_tests.js

```diff
@@ -261,27 +305,30 @@ describe('CRUD spec', function() {
   function executeInsertTest(scenarioTest, db, collection) {
     const args = scenarioTest.operation.arguments;
     const documents = args.document || args.documents;
-    let options = Object.assign({}, args);
+    let options = Object.assign({}, args.options);
     delete options.document;
     delete options.documents;
 
-    return collection[scenarioTest.operation.name](documents, options).then(
-      assertWriteExpectations(collection, scenarioTest.outcome)
-    );
+    const asserter = assertWriteExpectations(collection, scenarioTest.outcome);
+
+    return collection[scenarioTest.operation.name](documents, options).then(asserter, asserter);
```

Comment:
This used to return a Promise with only a `then` attached to it, allowing errors to propagate back up if something went wrong before the expectations were checked. Now you pass in the "asserter" to both resolve and reject, but don't pass in a method that checks for an error.

To me it seems like it would be easier to read this flow if we did something more like:

```
const promise = collection[scenarioTest.operation.name](documents, options);
if (scenario.hasError) {
  promise.catch(doErrorChecks);
}

return promise.then(asserter);
```